src/main/java/com/google/devtools/build/lib/sandbox/cgroups/controller/Controller.java

```diff
@@ -0,0 +1,43 @@
+package com.google.devtools.build.lib.sandbox.cgroups.controller;
+
+import com.google.devtools.build.lib.actions.ExecException;
+import com.google.devtools.build.lib.server.FailureDetails;
+
+import java.io.IOException;
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.lang.reflect.Proxy;
+import java.nio.file.Path;
+
+public interface Controller {
+    default boolean isLegacy() throws IOException {
+        return !getPath().resolve("cgroup.controllers").toFile().exists();
+    }
+
+    static <T extends Controller> T getDefault(Class<T> clazz) {
```

Comment:
I think it would be better to implement a simple NO-OP class to handle the case when cgroups are not available. 

See: https://github.com/bazelbuild/bazel/blob/d839db66ee86bc906a9789e99d11e0b85aba20be/src/main/java/com/google/devtools/build/lib/sandbox/CgroupsInfo.java#L241.

Reason being, we do want some control in the code when working with controllers where cgroups are not available (either due to permissions or setup) to handle this gracefully - e.g. ignore the flag entirely with a warning.

Reply:
Sure, I was considering adding a flag for the user to decide (e.g. `--cgroup_fail_if_unavailable`), but can be discussed/added separately